### PR TITLE
httpproxy: fix race on getting close notifier channel

### LINKED ISSUE #5267: Etcd 2.3.3 proxy panic - CloseNotify called after ServeHTTP finished
An etcd instance running 2.3.3 on linux has panicked this morning. The node was a proxy node and it failed with the following trace:

```
panic: net/http: CloseNotify called after ServeHTTP finished

goroutine 35455 [running]:
panic(0xbeaea0, 0xc8204262b0)
        /usr/local/go/src/runtime/panic.go:481 +0x3e6
net/http.(*response).CloseNotify(0xc8200721a0, 0x0)
        /usr/local/go/src/net/http/server.go:1535 +0x9d
github.com/coreos/etcd/proxy.(*reverseProxy).ServeHTTP.func1(0x7f623aa76f48, 0xc8200721a0, 0xc820444008, 0xc820197a40, 0xc820444160, 0xc8207de070)
        /home/gyuho/go/src/github.com/coreos/etcd/release/etcd/gopath/src/github.com/coreos/etcd/proxy/reverse.go:116 +0x39
created by github.com/coreos/etcd/proxy.(*reverseProxy).ServeHTTP
        /home/gyuho/go/src/github.com/coreos/etcd/release/etcd/gopath/src/github.com/coreos/etcd/proxy/reverse.go:122 +0xa1d
```

No special operation (cluster has been running for a while without any problem)

Do you have any hint?

Thanks a lot,

### PR DESCRIPTION
**Pull request by heyitsanthony**
Fixes #5267

### CODE REVIEW

File: proxy/httpproxy/reverse.go [hunk 1]
```diff
@@ -112,9 +112,10 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 	closeNotifier, ok := rw.(http.CloseNotifier)
```
Comment (xiang90) [inline proxy/httpproxy/reverse.go]: Can you explain more about the race? I failed to find the race.

Comment (heyitsanthony) [inline proxy/httpproxy/reverse.go]: The response writer is closed before the goroutine is scheduled. This nils out some internal state of the response writer so when the goroutine finally runs CloseNotify() it panics.

**Comment by xiang90** [comment]
LGTM.

**Comment by gyuho** [comment]
LGTM. Thanks.

**Comment by raoofm** [comment]
This should be back-ported right.

**Comment by xiang90** [comment]
@raoofm Yes. Next week.

**Comment by raoofm** [comment]
@xiang90 Thanks

**Pull request merged**